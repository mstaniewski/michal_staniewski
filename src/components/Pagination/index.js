import React, { useMemo } from "react";
import PropTypes from "prop-types";

const Pagination = ({ pagination: { totalPages, page }, onChange }) => {
  const pages = useMemo(() => new Array(totalPages).fill(null), [totalPages]);

  return (
    <nav aria-label="Data grid navigation">
      <ul className="pagination justify-content-end">
        <li
          onClick={() => page > 1 && onChange(page - 1)}
          className={`page-item ${page === 1 ? "disabled" : ""}`}
        >
          <span className="page-link" tabIndex="-1">
            Previous
          </span>
        </li>
        {pages.map((_, idx) => (
          <li
            key={`page-${idx + 1}`}
            onClick={() => onChange(idx + 1)}
            className={`page-item ${idx + 1 === page ? "active" : ""}`}
          >
            <span className="page-link">
              {idx + 1} <span className="sr-only">(current)</span>
            </span>
          </li>
        ))}
        <li
          onClick={() => page < totalPages && onChange(page + 1)}
          className={`page-item ${page === totalPages ? "disabled" : ""}`}
        >
          <span className="page-link">Next</span>
        </li>
      </ul>
    </nav>
  );
};

Pagination.defaultProps = {
  onChange: () => {},
  pagination: {
    page: 1,
    totalPages: 1
  }
};

Pagination.propTypes = {
  pagination: PropTypes.shape({
    totalPages: PropTypes.number.isRequired,
    page: PropTypes.number.isRequired
  }),
  onChange: PropTypes.func.isRequired
};

export default Pagination;
