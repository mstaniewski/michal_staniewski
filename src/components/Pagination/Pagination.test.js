import React from "react";
import Pagination from "./index";
import { shallow } from "enzyme";

const TEST_PAGINATION = {
  page: 1,
  totalPages: 2
};

describe("<Pagination />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<Pagination pagination={TEST_PAGINATION} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match proper number of pages", () => {
    const wrapper = shallow(<Pagination pagination={TEST_PAGINATION} />);
    // + 2 because of prev / next controls
    expect(wrapper.find(".page-item")).toHaveLength(
      TEST_PAGINATION.totalPages + 2
    );
  });

  it("should disable previous button when on first page", () => {
    const wrapper = shallow(<Pagination pagination={TEST_PAGINATION} />);
    expect(
      wrapper
        .find(".page-item")
        .first()
        .hasClass("disabled")
    ).toBeTruthy();
  });

  it("should disable previous button when on last page", () => {
    const wrapper = shallow(
      <Pagination
        pagination={{ ...TEST_PAGINATION, page: TEST_PAGINATION.totalPages }}
      />
    );
    expect(
      wrapper
        .find(".page-item")
        .last()
        .hasClass("disabled")
    ).toBeTruthy();
  });

  it("should add active class to current item", () => {
    const wrapper = shallow(
      <Pagination
        pagination={{ ...TEST_PAGINATION, page: TEST_PAGINATION.totalPages }}
      />
    );
    expect(
      wrapper
        .find(".page-item")
        .at(TEST_PAGINATION.totalPages)
        .hasClass("active")
    ).toBeTruthy();
  });
});
