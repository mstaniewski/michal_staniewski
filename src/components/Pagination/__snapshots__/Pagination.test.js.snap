// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should match snapshot 1`] = `ShallowWrapper {}`;
